Contest_API.md

```diff
@@ -497,14 +499,19 @@ are mentioned below.
 
 ### Types of endpoints
 
-The endpoints can be categorized into 3 groups as follows:
+The endpoints can be categorized into 4 groups as follows:
 
+  - Metadata: `/`, access
   - Configuration: contests, judgement-types, languages, problems,
     groups, organizations, teams, team-members;
   - Live data: state, submissions, judgements, runs, clarifications,
     awards, commentary;
   - Aggregate data: scoreboard, event-feed.
 
+Metadata is data about the API. These are not included in the event feed and
+are always required to be available. The access endpoint specifies which other
+endpoints are offered by the API. 
```

Comment:
The Extensibility section should be clear enough that additional un-spec'd properties are allowed.

Reply:
I meant it in the sense that all attributes currently specified in the `/` and `access` endpoints are required, but in the future we could specify optional attributes there. Do we already want to accomodate for that in the text and tables now?

Reply:
The fact that `/access` specifies endpoints by reference to a table in the spec means that it's not (really) possible to add extensions in the form of new endpoints. Do we want to fix that?

Reply:
> The fact that `/access` specifies endpoints by reference to a table in the spec means that it's not (really) possible to add extensions in the form of new endpoints. Do we want to fix that?

Yes, I think it would be good to somehow mention that you can provide unspecified endpoints or properties, and then you should also include these in the `access` endpoint. That's how I interpret the extensibility section.

But there's also the meta-point, that the `access` endpoint doesn't specify what properties are available in the meta-data endpoints `/` or `access` themselves. I guess we don't have to tackle that in this PR, but would be good to put on the list with scoreboard and event-feed endpoints to someone add to `access`.

Reply:
> Yes, I think it would be good to somehow mention that you can provide unspecified endpoints or properties, and then you should also include these in the `access` endpoint.

Right, but you can't actually do that. The values for the "type" property refers to the notification format table for valid values. Any unspecified endpoint that we want to include will, by definition, not be in that list, so it can't be added to the `access` endpoint.

> But there's also the meta-point, that the `access` endpoint doesn't specify what properties are available in the meta-data endpoints `/` or `access` themselves.

I don't think that's a big problem. Both of those endpoints are specified to be always required, and at least for now all properties must always be there.

Reply:
I made it more clear that all properties in `/` and `/access` are required.